src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/Runners/MonitorRunner.cs

```diff
@@ -130,6 +135,12 @@ public virtual async Task StartAsync(string command, string[] args, Cancellation
                 argsList.AddRange(args);
             }
 
+            if (ExplicitlySetConfigurationFilePath)
```

Comment:
Instead of overloading the use of the `UserSettingsFilePath`, please add another method similar to `WriteUserSettingsAsync` for writing out a configuration file that would be passed via the command line. You can write it into directory under `_tempDir` so that it is cleaned up when the runner is disposed.